api-controllers/FabrikBaseController.js

```diff
@@ -75,11 +77,11 @@ class FabrikBaseController extends BaseController {
               operationType === CONST.OPERATION_TYPE.CREATE && res.statusCode === CONST.HTTP_STATUS_CODE.CONFLICT || // PutInstance => unlock in case of 409
               operationType === CONST.OPERATION_TYPE.DELETE && res.statusCode === CONST.HTTP_STATUS_CODE.GONE // DeleteInstance => unlock in case of 410
             ) {
-              return lockManager.unlock(req.params.instance_id)
+              return lockManager.unlock(req.params.instance_id, req.lockId)
```

Comment:
The idea is that whoever acquires lock should have the lockId as it is needed in PATCH call to handle race conditions.
So either it is passed to `lockManager.unlock()` (1) or unlock can get it from the resource who is trying to unlock(2)
In 2nd case also then we will have to pass `resourceGroup` and `resourceType` and `resourceId` as we can't rely on `spec.options.lockedResourceDetails` of lock resource as it might have changed (race condition)